hearpreprocess/pipeline.py

```diff
@@ -748,19 +748,29 @@ def run(self):
         # minutes or the timestamp embeddings will explode
         sample_duration = self.task_config["sample_duration"]
 
-        if (
-            "max_task_duration_by_split" in self.task_config
-            and self.split in self.task_config["max_task_duration_by_split"]
-        ):
-            max_split_duration = self.task_config["max_task_duration_by_split"][
-                self.split
-            ]
+        # Get the max split duration from the task config.
+        # If not defined use defaults from MAX_TASK_DURATION_BY_SPLIT
+        if "max_task_duration_by_split" in self.task_config:
+            # Check if all the splits specified in the
+            # max_task_duration_by_split are in SPLITS
+            assert set(max_task_duration_by_split.keys()) <= set(SPLITS)
```

Comment:
`max_task_duration_by_split` isn't a variable. You need `self.task_config['max_task_duration_by_split']`

Reply:
Yeah. This simplification is not complete. Resolved this now. Thanks